examples/nextjs-boilerplate/src/context/canvas.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {
+  useState,
+  createContext,
+  useEffect,
+  useContext,
+  useRef,
+} from 'react';
+import {
+  createCanvasClient,
+  type CanvasClient,
+  type CanvasInterface,
+} from '@dscvr-one/canvas-client-sdk';
+import { registerCanvasWallet } from '@dscvr-one/canvas-wallet-adapter';
+import { validateHostMessage } from '@/api/dscvr';
+
+type CanvasContextType = {
+  client?: CanvasClient;
+  user?: CanvasInterface.Lifecycle.User;
+  content?: CanvasInterface.Lifecycle.Content;
+};
+
+const CanvasContext = createContext<CanvasContextType>({});
+
+export const CanvasProvider = ({ children }: { children: React.ReactNode }) => {
+  const [canvasContext, setCanvasContext] = useState<CanvasContextType>({});
+  const resizeObserverRef = useRef<ResizeObserver>();
+
+  const initialize = async () => {
+    const canvasClient = createCanvasClient();
+    setCanvasContext({ client: canvasClient });
+
+    if (!canvasClient) return;
+    registerCanvasWallet(canvasClient);
+
+    try {
+      const response = await canvasClient.ready();
+      const isValidResponse = await validateHostMessage(response);
+      if (isValidResponse) {
+        setCanvasContext({
+          client: canvasClient,
+          user: response.untrusted.user,
+          content: response.untrusted.content,
+        });
+      }
+    } catch (error) {
+      console.error('Failed to initialize canvas', error);
+    }
+  };
+
+  useEffect(() => {
+    initialize();
+
+    return () => {
+      canvasContext.client?.destroy();
+    };
+  }, []);
+
+  useEffect(() => {
+    if (canvasContext.client) {
+      resizeObserverRef.current = new ResizeObserver(() =>
+        canvasContext.client?.resize(),
+      );
+      resizeObserverRef.current.observe(document.body);
+
+      return () => {
+        resizeObserverRef.current?.disconnect();
+      };
+    }
+  }, [canvasContext, canvasContext.client]);
```

Comment:
I think here you dont need canvasContext but just canvasContext.client

Reply:
We don't thanks